### PR TITLE
Adds fixture factories plugin to the testing documentation.

### PR DESCRIPTION
**Pull request by pabloelcolombiano**
As discussed earlier this week per email, this is a proposal for having the fixture factories package mentioned in the documentation.

### CODE REVIEW

File: en/development/testing.rst [hunk 1]
```diff
@@ -720,6 +720,46 @@ name::
 In the above example, both fixtures would be loaded from
 ``tests/Fixture/Blog/``.
 
+Fixture Factories
+-----------------
+
+While your application grows, so do the number and the size of your test fixtures. You might find it difficult
+to maintain them and to keep track of their content.
+The `fixture factories plugin<https://github.com/vierge-noire/cakephp-fixture-factories>`_ proposes an
```
Comment (markstory) [inline en/development/testing.rst]:
```suggestion
The `fixture factories plugin <https://github.com/vierge-noire/cakephp-fixture-factories>`_ proposes an
```

File: en/development/testing.rst [hunk 2]
```diff
@@ -720,6 +720,46 @@ name::
...
+
+The plugin uses its own `phpunit listener <https://github.com/vierge-noire/cakephp-test-suite-light>`_,
+which will perform the following actions:
+
+#. Run migrations once on the test DB `(see the migrator) <https://github.com/vierge-noire/cakephp-test-migrator>`_.
+#. Truncate dirty tables before each test.
+#. Run tests.
+
+The following command will help you bake your factories:
+``bin/cake bake fixture_factory -h``
```
Comment (markstory) [inline en/development/testing.rst]:
```suggestion
The following command will help you bake your factories::

    bin/cake bake fixture_factory -h
```

File: en/development/testing.rst [hunk 3]
```diff
@@ -720,6 +720,46 @@ name::
...
+
+Once your factories are
+`tuned <https://github.com/vierge-noire/cakephp-fixture-factories/blob/master/docs/factories.md>`_,
+you are ready to create test fixtures in no time.
+
+Unnecessary interaction with the database will slow down your tests, as well as your application.
+You can create test fixtures without persisting them, which can be useful for
+testing methods without DB interaction.
+``$article = ArticleFactory::make()->getEntity();``
```
Comment (markstory) [inline en/development/testing.rst]:
```suggestion
testing methods without DB interaction::

    $article = ArticleFactory::make()->getEntity();
```

File: en/development/testing.rst [hunk 4]
```diff
@@ -720,6 +720,46 @@ name::
...
+
+In order to persist:
+``$article = ArticleFactory::make()->persist();``
```
Comment (markstory) [inline en/development/testing.rst]:
```suggestion
In order to persist::

    $article = ArticleFactory::make()->persist();
```

File: en/development/testing.rst [hunk 5]
```diff
@@ -720,6 +720,46 @@ name::
...
+
+The factories help creating associated fixtures too.
+Assuming that articles belongs to many authors, we can now for example
+create 5 articles with each 2 authors:
+
+``$articles = ArticleFactory::make(5)->with('Authors', 2)->getEntities();``
```
Comment (markstory) [inline en/development/testing.rst]:
```suggestion
create 5 articles with each 2 authors::

    $articles = ArticleFactory::make(5)->with('Authors', 2)->getEntities();
```

File: en/development/testing.rst [hunk 6]
```diff
@@ -720,6 +720,46 @@ name::
...
+
+Note that the fixture factories do not require any fixture creation or declaration. Still, they are fully
+compatible with the fixtures that come with cakephp. You will find additional insights
+and documentation `here<https://github.com/vierge-noire/cakephp-fixture-factories>`_.
```
Comment (markstory) [inline en/development/testing.rst]:
```suggestion
and documentation `here <https://github.com/vierge-noire/cakephp-fixture-factories>`_.
```

File: fr/development/testing.rst [hunk 7]
```diff
@@ -717,6 +717,45 @@ la fixture::
...
+
+Le plugin utilise son propre `phpunit listener <https://github.com/vierge-noire/cakephp-test-suite-light>`_,
+qui effectue les actions suivantes:
+
+#. Faire tourner les migrations `(description ici) <https://github.com/vierge-noire/cakephp-test-migrator>`_.
+#. Tronquer les tables utilisées au préalable avant chaque test.
+#. Lancer les tests.
+
+La commande bake suivante vous assistera pour créer vos factories:
+``bin/cake bake fixture_factory -h``
```
Comment (markstory) [inline fr/development/testing.rst]:
```suggestion
La commande bake suivante vous assistera pour créer vos factories::

    bin/cake bake fixture_factory -h
```

File: fr/development/testing.rst [hunk 8]
```diff
@@ -717,6 +717,45 @@ la fixture::
...
+Il est possible de créer des fixtures sans les insérer. Ceci est utile lorsque vous testez des méthodes
+qui n'intéragissent pas avec la base de donnée.
+``$article = ArticleFactory::make()->getEntity();``
+
+Pour insérer dans la base de donnée:
+``$article = ArticleFactory::make()->persist();``
+
+En supposant que les articles appartiennent à plusieurs auteurs, il est possible de créer 5 articles ayant chacun
+2 auteurs de la manière suivante:
+``$articles = ArticleFactory::make(5)->with('Authors', 2)->getEntities();``
```
Comment (markstory) [inline fr/development/testing.rst]:
```suggestion
qui n'intéragissent pas avec la base de donnée::

    $article = ArticleFactory::make()->getEntity();

Pour insérer dans la base de donnée::

    $article = ArticleFactory::make()->persist();

En supposant que les articles appartiennent à plusieurs auteurs, il est possible de créer 5 articles ayant chacun
2 auteurs de la manière suivante::

    $articles = ArticleFactory::make(5)->with('Authors', 2)->getEntities();
```

File: fr/development/testing.rst [hunk 9]
```diff
@@ -717,6 +717,45 @@ la fixture::
...
+
+Notez que bien que les factories ne nécessitent ni la création, ni la déclaration de fixtures, elles sont
+parfaitement compatibles avec ces dernières. Pour plus de détails,
+rendez-vous `ici<https://github.com/vierge-noire/cakephp-fixture-factories>`_.
```
Comment (markstory) [inline fr/development/testing.rst]:
```suggestion
rendez-vous `ici <https://github.com/vierge-noire/cakephp-fixture-factories>`_.
```

**Review by markstory — COMMENTED** [review]
There are a few formatting changes to make.

File: en/development/testing.rst [hunk 10]
```diff
@@ -720,6 +720,46 @@ name::
 In the above example, both fixtures would be loaded from
 ``tests/Fixture/Blog/``.
 
+Fixture Factories
+-----------------
+
+While your application grows, so do the number and the size of your test fixtures. You might find it difficult
```
Comment (othercorey) [inline en/development/testing.rst]:
```suggestion
As your application grows, so does the number and the size of your test fixtures. You might find it difficult
```

File: en/development/testing.rst [hunk 11]
```diff
@@ -720,6 +720,46 @@ name::
...
+to maintain them and to keep track of their content.
+The `fixture factories plugin<https://github.com/vierge-noire/cakephp-fixture-factories>`_ proposes an
+alternative for large sized applications.
+
+The plugin uses its own `phpunit listener <https://github.com/vierge-noire/cakephp-test-suite-light>`_,
```
Comment (othercorey) [inline en/development/testing.rst]:
```suggestion
The plugin uses its own `phpunit listener <https://github.com/vierge-noire/cakephp-test-suite-light>`_
```

File: en/development/testing.rst [hunk 12]
```diff
@@ -720,6 +720,46 @@ name::
...
+#. Run tests.
+
+The following command will help you bake your factories:
+``bin/cake bake fixture_factory -h``
+
+Once your factories are
+`tuned <https://github.com/vierge-noire/cakephp-fixture-factories/blob/master/docs/factories.md>`_,
+you are ready to create test fixtures in no time.
+
+Unnecessary interaction with the database will slow down your tests, as well as your application.
```
Comment (othercorey) [inline en/development/testing.rst]:
```suggestion
Unnecessary interaction with the database will slow down your tests as well as your application.
```

File: en/development/testing.rst [hunk 13]
```diff
@@ -720,6 +720,46 @@ name::
...
+You can create test fixtures without persisting them, which can be useful for
```
Comment (othercorey) [inline en/development/testing.rst]:
```suggestion
You can create test fixtures without persisting them which can be useful for
```

File: en/development/testing.rst [hunk 14]
```diff
@@ -720,6 +720,46 @@ name::
...
+testing methods without DB interaction.
+``$article = ArticleFactory::make()->getEntity();``
+
+In order to persist:
+``$article = ArticleFactory::make()->persist();``
+
+The factories help creating associated fixtures too.
+Assuming that articles belongs to many authors, we can now for example
+create 5 articles with each 2 authors:
```
Comment (othercorey) [inline en/development/testing.rst]:
```suggestion
Assuming that articles belongs to many authors, we can now, for example,
create 5 articles each with 2 authors:
```

**Review by pabloelcolombiano — COMMENTED** [review]
Thanks for the review, all suggestions were applied.

**Comment by markstory** [comment]
Thank you 👍

**Pull request merged**